src/libmongoc/src/mongoc/mongoc-cursor.c

```diff
@@ -1070,11 +1073,17 @@ _mongoc_cursor_run_command (mongoc_cursor_t *cursor,
 
       mongoc_server_stream_cleanup (server_stream);
 
-      server_stream = mongoc_cluster_stream_for_reads (&cursor->client->cluster,
-                                                       cursor->read_prefs,
-                                                       cursor->client_session,
-                                                       reply,
-                                                       &cursor->error);
+      BSON_ASSERT (cursor->is_aggr_with_write_stage ==
```

Comment:
Confirming: is it appropriate to be asserting this condition vs. treating it as a recoverable command failure?

Reply:
No, this is not corresponding to a test failure, but the enforcement of being retryable is spread throughout the codebase enough that I feel it should be guarded against with an assert to prevent future modifications from accidentally retrying an aggregate+write operation. The assert isn't to check that the user doesn't do something wrong, but to assert that the driver code itself doesn't do something wrong. My reasoning is that retrying a write could cause a double-write, which may corrupt user data, so it would be better to halt than allow it to pass through.